nemoguardrails/library/gliner/request.py

```
@@ -74,29 +80,68 @@ async def gliner_request(
                 )
+
+            if not use_nim_format:
+                return raw
+
+            # Unwrap chat completions envelope and normalize entity field names.
+            # NIM uses: text, label, start, end
+            # Custom server uses: value, suggested_label, start_position, end_position
+            try:
+                nim_data = json.loads(raw["choices"][0]["message"]["content"])
+            except (KeyError, IndexError, json.JSONDecodeError) as e:
+                raise ValueError(f"Failed to parse NIM response content: {e}")
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)